src/baseService.ts

```diff
@@ -263,17 +263,17 @@ export class BaseService {
   @nonHTTPRequestMethod
   private _resolveQuery(methodName: string, args: any[]): any {
     const meta = this.__meta__;
-    const query = meta[methodName].query || {};
+    const query = { ...meta[methodName].query };
     const queryParams = meta[methodName].queryParams;
     for (const pos in queryParams) {
-      if (queryParams[pos]) {
+      if (queryParams[pos] && args[pos] !== undefined) {
```

Comment:
### Issue 2 - sending undefined query value

Once leak was fixed, optional params with no value specified would always be present as keys in query  config 

```
{ 
    myoptionalparam: undefined, 
    myrequiredparam: 'hello' 
}

This isn't desirable as given the query string `/something?myoptionalparam=undefined` a server would read this as a string with value of `"undefined"`
 
For headers this was worse as errored out due to trying to call `setHeader(undefined)`

Avoiding adding unspecified optional query and header args avoids this, so they don't get sent with the request at all